### ISSUE TITLE
Traceback from bodhi-dequeue-stable

### ISSUE THREAD

**Issue by bowlofeggs** [fedora-infra/bodhi#2756]
@nirik sent me a traceback that seems to be coming from ```bodhi-dequeue-stable```:

```
Unable to send mail
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/bodhi/server/mail.py", line
391, in _send_mail
    smtp.sendmail(from_addr, [to_addr], body)
  File "/usr/lib64/python3.7/smtplib.py", line 861, in sendmail
    (code, resp) = self.mail(from_addr, esmtp_opts)
  File "/usr/lib64/python3.7/smtplib.py", line 534, in mail
    self.putcmd("mail", "FROM:%s%s" % (quoteaddr(sender), optionlist))
  File "/usr/lib64/python3.7/smtplib.py", line 150, in quoteaddr
    displayname, addr = email.utils.parseaddr(addrstring)
  File "/usr/lib64/python3.7/email/utils.py", line 212, in parseaddr
    addrs = _AddressList(addr).addresslist
  File "/usr/lib64/python3.7/email/_parseaddr.py", line 500, in __init__
    self.addresslist = self.getaddrlist()
  File "/usr/lib64/python3.7/email/_parseaddr.py", line 253, in getaddrlist
    ad = self.getaddress()
  File "/usr/lib64/python3.7/email/_parseaddr.py", line 263, in getaddress
    self.gotonext()
  File "/usr/lib64/python3.7/email/_parseaddr.py", line 236, in gotonext
    if self.field[self.pos] in self.LWS + '\n\r':
TypeError: 'in <string>' requires string as left operand, not int
```

**Comment by bowlofeggs** [comment]
It seems the composer also crashes with a similar if not identical traceback:

```
[2018-11-20 14:34:25][fedmsg.consumers    INFO]f28-updates Sending notifications
[2018-11-20 14:34:26][fedmsg.consumers    INFO]f28-updates Sending stable update announcements
[2018-11-20 14:34:27][bodhi.server    INFO]f28-updates Sending mail to b'package-announce@fpo': b'Fedora 28 Update: slibtool-0.5.28-1.fc28'

[2018-11-20 14:34:27][bodhi.server   ERROR]f28-updates Unable to send mail
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/bodhi/server/mail.py", line 391, in _send_mail
    smtp.sendmail(from_addr, [to_addr], body)
  File "/usr/lib64/python3.7/smtplib.py", line 861, in sendmail
    (code, resp) = self.mail(from_addr, esmtp_opts)
  File "/usr/lib64/python3.7/smtplib.py", line 534, in mail
    self.putcmd("mail", "FROM:%s%s" % (quoteaddr(sender), optionlist))
  File "/usr/lib64/python3.7/smtplib.py", line 150, in quoteaddr
    displayname, addr = email.utils.parseaddr(addrstring)
  File "/usr/lib64/python3.7/email/utils.py", line 212, in parseaddr
    addrs = _AddressList(addr).addresslist
  File "/usr/lib64/python3.7/email/_parseaddr.py", line 500, in __init__
    self.addresslist = self.getaddrlist()
  File "/usr/lib64/python3.7/email/_parseaddr.py", line 253, in getaddrlist
    ad = self.getaddress()
  File "/usr/lib64/python3.7/email/_parseaddr.py", line 263, in getaddress
    self.gotonext()
  File "/usr/lib64/python3.7/email/_parseaddr.py", line 236, in gotonext
    if self.field[self.pos] in self.LWS + '\n\r':
TypeError: 'in <string>' requires string as left operand, not int
```

**Comment by bowlofeggs** [comment]
This seems relevant: https://bugs.python.org/issue20084

**Comment by bowlofeggs** [comment]
Well here's a simple reproducer:

```
$ python3
Python 3.7.1 (default, Nov  5 2018, 14:07:04) 
[GCC 8.2.1 20181011 (Red Hat 8.2.1-4)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import smtplib
>>> e = smtplib.SMTP('my.smtp.server')
>>> e.sendmail(b'package-announce@fedoraproject.org', [b'me@mydomain.com'], b'Subject: sup\n\nhello')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.7/smtplib.py", line 861, in sendmail
    (code, resp) = self.mail(from_addr, esmtp_opts)
  File "/usr/lib64/python3.7/smtplib.py", line 534, in mail
    self.putcmd("mail", "FROM:%s%s" % (quoteaddr(sender), optionlist))
  File "/usr/lib64/python3.7/smtplib.py", line 150, in quoteaddr
    displayname, addr = email.utils.parseaddr(addrstring)
  File "/usr/lib64/python3.7/email/utils.py", line 212, in parseaddr
    addrs = _AddressList(addr).addresslist
  File "/usr/lib64/python3.7/email/_parseaddr.py", line 500, in __init__
    self.addresslist = self.getaddrlist()
  File "/usr/lib64/python3.7/email/_parseaddr.py", line 253, in getaddrlist
    ad = self.getaddress()
  File "/usr/lib64/python3.7/email/_parseaddr.py", line 263, in getaddress
    self.gotonext()
  File "/usr/lib64/python3.7/email/_parseaddr.py", line 236, in gotonext
    if self.field[self.pos] in self.LWS + '\n\r':
TypeError: 'in <string>' requires string as left operand, not int
```

**Comment by bowlofeggs** [comment]
The [docs](https://docs.python.org/3.7/library/smtplib.html#smtplib.SMTP.sendmail) state that the from_addr and to_addr should be strings, so perhaps it is wrong for us to convert them to ```bytes```.

**Comment by bowlofeggs** [comment]
The fix for this issue is now released with Bodhi 3.11.1 and deployed to Fedora's production instance.
